aws/sqs_event_manager/sqs_event_manager/app.py

```diff
@@ -122,48 +122,48 @@ def process_message(record: Dict) -> Dict[str, Union[str, bool]]:
     result = {
         'itemIdentifier': record.get('messageId') or 'unknown',
         'status': 'unknown',
         'error': True
     }
     try:
         message = AWSSNSMessage(record)
+        auth_token = sqs_event_manager_config.get('auth_token') or ''
         if not message.action:
             result['status'] = 'No action defined in SNS message'
             logger.error(result['status'])
         elif message.action == 'entitlement-updated':
-            # Notify SCC of customer entitlement change for the given product
-            customer_id = message.customer_id
-            product_code = message.product_code
-            logger.info(
-                'requesting entitlements for customer {} and product {}'.format(
-                    customer_id, product_code
+            endpoint_url = sqs_event_manager_config['entitlement_change_url']
+            result.update(
+                send_to_scc(
+                    entitlement_updated(message), endpoint_url, auth_token
                 )
             )
-            entitlements = AWSCustomerEntitlement(
-                customer_id, product_code
+        elif message.action == 'subscribe-success':
+            endpoint_url = sqs_event_manager_config['subscribe_success_url']
+            result.update(
+                send_to_scc(
+                    subscription_success(message), endpoint_url, auth_token
+                )
             )
-            request_data = {
-                'customerIdentifier': customer_id,
-                'marketplaceIdentifier': 'AWS',
-                'productCode': product_code,
-                'entitlements': entitlements.get_entitlements()
-            }
-            sqs_event_manager_config = Defaults.get_sqs_event_manager_config()
-            headers = {
-                'Content-Type': 'application/json'
-            }
-            auth_token = sqs_event_manager_config.get('auth_token')
-            if auth_token:
-                headers['Authorization'] = f'Bearer {auth_token}'
-            http_post_response = requests.post(
-                sqs_event_manager_config['entitlement_change_url'],
-                data=request_data,
-                headers=headers
+        elif message.action == 'unsubscribe-success':
+            endpoint_url = sqs_event_manager_config['unsubscribe_success_url']
+            result.update(
+                send_to_scc(
+                    subscription_removed(message), endpoint_url, auth_token
+                )
             )
-            http_post_response.raise_for_status()
-            result['status'] = format(http_post_response.status_code)
+        elif message.action == 'subscribe-fail':
+            endpoint_url = sqs_event_manager_config['subscribe_fail_url']
+            result.update(
+                send_to_scc(
+                    subscription_failed(message), endpoint_url, auth_token
+                )
+            )
+        elif message.action == 'unsubscribe-pending':
```

Comment:
I think if unsubscribe is pending you are supposed to bill any outstanding usage at this time. Not sure how we would want to handle that workflow?

Reply:
We have agreed that the potentially last hour is free, at least for now.